api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgeConfigSpec defines the desired state of EdgeConfig
+type EdgeConfigSpec struct {
+	// The ansible playbook command to execute
+	EdgePlaybook *EdgePlaybookSpec `json:"edgePlaybook,omitempty"`
+
+	// EdgeDeviceGroupName in which this EdgeConfig is applied
+	EdgeDeviceGroupName string `json:"edgeDeviceGroupName,omitempty"`
```

Comment:
With a single group name where won't be an option to support multiple groups by the same config resource.

To support multiple groups by the same config, we can go with [same](https://github.com/project-flotta/flotta-operator/pull/180/files#diff-42d3e0337d69da396a4fc9def6162460e18795c57734394a89ec10a66c788630R80) solution for managing devices by a device group: Introducing a label `flotta/configured_by: edge_config_name` on all of the groups be managed by it.

Hence, from CRD pov, no attributed is needed. It will be addressed by the controller.

Reply:
I don't know what is better. labels or specifying names in spec. However, I think it is better to do it the same as EdgeDevice and EdgeWorkload.

Reply:
I removed the `EdgeConfigName` and I added the `DeviceIDs` field to manage device without a label group.
WDYT?